### ISSUE TITLE
DVD discs not played

### ISSUE THREAD

**Issue by Ricky-Tigg** [mpv-player/mpv#10781]
**v.0.34.1** | Default configuration **unmodified** | **No script added**

Hello  On my laptop, which has a DVD drive embbeded, DVD discs with relevant region code can be played in VLC, in Windows 64 bit. Discs' contents start to play automatically as intended as soon as discs are inserted. Since then for conveniance i uninstalled VLC in order to test MPV. That latter is installed under C:\Program Files\mpv.

Insertions of disc do not trigger movies to be played automatically despite being set accordingly as such. DVD drive icon dragged into MPV screen scene causes DVD disc titles to be reported in MPV's title bar as _E:\  - mpv_. From that step the application remains running with an empty screen scene about four seconds then quits by itself, hence discs' contents never get played.

**Comment by Ricky-Tigg** [comment]
And here it is:

```
>cd C:\Program Files\mpv
>mpv dvd://
libdvdread: Could not open D: with libdvdcss.
libdvdread: Can't open D: for reading
libdvdnav: vm: failed to open/read the DVD
[dvdnav] Couldn't open DVD device: D:
Failed to open dvd://.

Exiting... (Errors when loading file)

>mpv E:\
[ifo_dvdnav] .IFO detected. Redirecting to dvd://
libdvdread: Could not open E:\VIDEO_TS/ with libdvdcss.
libdvdread: Can't open E:\VIDEO_TS/ for reading
libdvdread: Device E:\VIDEO_TS/ inaccessible, CSS authentication not available.
libdvdnav: Unable to open device file E:\VIDEO_TS/.
libdvdnav: vm: dvd_read_name failed
libdvdnav: DVD disk reports itself with Region mask 0x00fd0000. Regions: 02
libdvdnav: decoder.c: [WARNING, unknown bits: 1000000000080000]
libdvdnav: decoder.c: [WARNING, unknown bits: 1000000000000000]
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
(...)
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
[ifo_dvdnav] Selecting title 5.
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
[ifo_dvdnav] DVDNAV, switched to title: 4
[ifo_dvdnav] Error getting next block from DVD 1 (Error reading NAV packet.)
(...)
[ifo_dvdnav] Error getting next block from DVD 1 (Error reading NAV packet.)
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
(...)
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
No video or audio streams selected.

Exiting... (Errors when loading file)
```

**Comment by Ricky-Tigg** [comment]
This sets a partial amount of functionalities to be enabled. The movie is started in preselected language **without the exhibition of existing options**.  Hence, for example, the language is not available for selection.
```
>mpv dvd:// --dvd-device=E:
libdvdnav: Unable to open device file E:.
libdvdnav: vm: dvd_read_name failed
libdvdnav: DVD disk reports itself with Region mask 0x00fd0000. Regions: 02
libdvdread: Attempting to retrieve all CSS keys
libdvdread: This can take a _long_ time, please be patient
libdvdread: Get key for /VIDEO_TS/VIDEO_TS.VOB at 0x000001e6
libdvdread: Elapsed time 0
(...)
libdvdread: Found 16 VTS's
libdvdread: Elapsed time 0
libdvdnav: decoder.c: [WARNING, unknown bits: 1000000000080000]
libdvdnav: decoder.c: [WARNING, unknown bits: 1000000000000000]
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
(...)
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
[dvdnav] Selecting title 5.
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
[dvdnav] DVDNAV, switched to title: 4
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
(...)
libdvdread: CHECK_VALUE failed in /__w/mpv-winbuild-cmake/mpv-winbuild-cmake/src_packages/libdvdread/src/ifo_read.c:917 for pgc->cell_playback_offset != 0
 (+) Video --vid=1 (mpeg2video 720x576 25.000fps)
 (+) Audio --aid=1 --alang=sv (ac3 6ch 48000Hz)
     Audio --aid=2 --alang=da (ac3 6ch 48000Hz)
     Audio --aid=3 --alang=fi (ac3 6ch 48000Hz)
AO: [wasapi] 48000Hz stereo 2ch float
VO: [gpu] 720x576 => 1024x576 yuv420p
AV: 00:00:24 / 01:28:58 (0%) A-V:  0.000
```

**Comment by thebombzen** [comment]
`mpv` doesn't support DVD menus. You can always select the audio language with `--alang`

**Comment by Ricky-Tigg** [comment]
Since your comment i discovered that the **selection of audio track** is supported inside the interface; here shown as _1/3_.

![mpv_interface_audio_track_selection](https://user-images.githubusercontent.com/26058215/198274016-cc148362-96a6-456a-9f94-a95d064ec709.PNG)

I learned here that MPV has support of command-line – illustration of a valid command: `mpv dvd:// --dvd-device=E: --vid=1 --aid=1 --alang=sv`. Nevertheless as you noticed it by yourself, that was not the purpose of the present report to investigate that aspect.

Closing now as _completed_, assuming it is no accident, indicates that MPV is designed with **no support for playing contents of DVD discs** while DVD drive is dragged as **file path**. P Lease note that using a product that is tied to the interaction of the user of command-line in order to play DVD discs is not a suitable option for users like me who seek for minimalist and efficient way in this regard, not speaking of lowering the convenience that such use involves. Bringing useless complexity illustrates well what a **bad program design** leads to. The situation may not as desperate as it looks like for MPV. There is still the happy ones; those who value the use of command-line for the solely purpose of relying on command-line. Those ones shall dully value to have to run the required following commands each time they wish to play a DVD disc.
```
>cd C:\Program Files\[MPV_program_directory]
>mpv dvd:// --dvd-device=E:
```
For me – No thank you.

**Comment by Ricky-Tigg** [comment]
Since your comment i discovered that the **selection of audio track** is supported inside the interface; here shown as _1/3_.

![mpv_interface_audio_track_selection](https://user-images.githubusercontent.com/26058215/198274016-cc148362-96a6-456a-9f94-a95d064ec709.PNG)

I learned here that MPV has support of command-line – illustration of a valid command: `mpv dvd:// --dvd-device=E: --vid=1 --aid=1 --alang=sv`. Nevertheless as you noticed it by yourself, that was not the purpose of the present report to investigate that aspect.

Closing now as _completed_, assuming it is no accident, indicates that MPV is designed with **no support for playing contents of DVD discs** while DVD drive is dragged as **file path**. P Lease note that using a product that is tied to the interaction of the user of command-line in order to play DVD discs is not a suitable option for users like me who seek for minimalist and efficient way in this regard, not speaking of lowering the convenience that such use involves. Bringing useless complexity illustrates well what a **bad program design** leads to. Nevertheless the current situation for MPV may not be as desperate as it looks like. There is still the happy ones; those who value the use of command-line for the solely purpose of relying on command-line. Those ones shall dully value to have to run the required following commands each time they wish to play a DVD disc.
```
>cd C:\Program Files\[MPV_program_directory]
>mpv dvd:// --dvd-device=E:
```
For me – No thank you.
